projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts

```diff
@@ -1,12 +1,56 @@
-import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  OnInit,
+} from '@angular/core';
 import { NavigationNode } from './navigation-node.model';
 
+const COLUMN_SIZE = 10;
+
 @Component({
   selector: 'cx-navigation-ui',
   templateUrl: './navigation-ui.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class NavigationUIComponent {
+export class NavigationUIComponent implements OnInit {
   @Input() dropdownMode = 'list';
   @Input() node: NavigationNode;
+  columns;
+
+  getColumns(children) {
+    const columns = new Array();
+
+    // Iterate subcategories
+    for (let i = 0; i < children.length; i++) {
+      // Column header node for each subcategory
+      const columnHeader: NavigationNode = {
+        isHeader: true,
+        title: children[i].title,
+        url: children[i].url,
+      };
+
+      // Split subcategory items and header into columns
+      if (children[i].children) {
+        const clonedChildren = children[i].children.slice(0);
```

Comment:
I would change that to spread operator.
```suggestion
        const clonedChildren = [columnHeader, ...children[i].children];
```

Reply:
You could even add columnHeader in the spread operator.